File: src/C#/CreationalPatterns/AbstractFactory/EraStyles/Products/MedievalHouse.cs
using EraStyles.Abstraction;
using EraStyles.Constants;
using System;
using System.Collections.Generic;
using System.Text;

namespace EraStyles.Products
{
    public class MedievalHouse : EraObject
    {
        public override void ShowDetails()
        {
            Console.WriteLine(string.Format(GlobalConstants.HouseMessage, "Medieval"));
        }
    }
}

File: src/JavaScript/CreationalPatterns/AbstractFactory/EraStyles/products/medievalHouse.js

const EraObject = require("../abstraction/eraObject");
const GlobalConstants = require("../constants/globalConstants");

class MedievalHouse extends EraObject {
    showDetails() {
        console.log(GlobalConstants.HouseMessage.replace("{0}", "Medieval"));
    }
}

module.exports = MedievalHouse;